### ISSUE TITLE
Game crashes when seaching in the creative inventory [3.0.0]

### ISSUE THREAD

**Issue by aninusmuffin** [tazta2ra/bluelotus#3]
Seems to be a problem with JEI + EMI + Modernfix

[mclo.gs](https://mclo.gs/RdSf0Lf)

```
---- Minecraft Crash Report ----
// On the bright side, I bought you a teddy bear!
Time: 2023-06-06 22:05:26
Description: charTyped event handler
java.lang.ClassCastException: class dev.emi.emi.jemi.runtime.JemiIngredientFilter cannot be cast to class mezz.jei.gui.ingredients.IngredientFilterApi (dev.emi.emi.jemi.runtime.JemiIngredientFilter and mezz.jei.gui.ingredients.IngredientFilterApi are in unnamed module of loader org.quiltmc.loader.impl.launch.knot.KnotClassLoader @4cc0edeb)
	at org.embeddedt.modernfix.searchtree.JEIBackedSearchTree.method_4810(JEIBackedSearchTree.java:57)
	at net.minecraft.class_481.method_2464(class_481.java:451)
	at net.minecraft.class_481.method_25400(class_481.java:383)
	at dev.architectury.impl.fabric.ScreenInputDelegate$DelegateScreen.method_25400(ScreenInputDelegate.java:42)
	at net.minecraft.class_309.method_1458(class_309.java:448)
	at net.minecraft.class_437.method_25412(class_437.java:489)
	at net.minecraft.class_309.method_1457(class_309.java:448)
	at net.minecraft.class_309.method_22677(class_309.java:463)
	at net.minecraft.class_1255.execute(class_1255.java:103)
	at net.minecraft.class_309.method_22675(class_309.java:463)
	at org.lwjgl.glfw.GLFWCharModsCallbackI.callback(GLFWCharModsCallbackI.java:46)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:197)
	at net.minecraft.class_1041.method_15998(class_1041.java:315)
	at net.minecraft.class_310.method_1523(class_310.java:1205)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:527)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------
-- Head --
Thread: Render thread
Stacktrace:
	at org.embeddedt.modernfix.searchtree.JEIBackedSearchTree.method_4810(JEIBackedSearchTree.java:57)
	at net.minecraft.class_481.method_2464(class_481.java:451)
	at net.minecraft.class_481.method_25400(class_481.java:383)
	at dev.architectury.impl.fabric.ScreenInputDelegate$DelegateScreen.method_25400(ScreenInputDelegate.java:42)
	at net.minecraft.class_309.method_1458(class_309.java:448)
	at net.minecraft.class_437.method_25412(class_437.java:489)
	at net.minecraft.class_309.method_1457(class_309.java:448)
	at net.minecraft.class_309.method_22677(class_309.java:463)
	at net.minecraft.class_1255.execute(class_1255.java:103)
	at net.minecraft.class_309.method_22675(class_309.java:463)
	at org.lwjgl.glfw.GLFWCharModsCallbackI.callback(GLFWCharModsCallbackI.java:46)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:197)
-- Affected screen --
Details:
	Screen name: net.minecraft.class_481
Stacktrace:
	at net.minecraft.class_437.method_25412(class_437.java:489)
	at net.minecraft.class_309.method_1457(class_309.java:448)
	at net.minecraft.class_309.method_22677(class_309.java:463)
	at net.minecraft.class_1255.execute(class_1255.java:103)
	at net.minecraft.class_309.method_22675(class_309.java:463)
	at org.lwjgl.glfw.GLFWCharModsCallbackI.callback(GLFWCharModsCallbackI.java:46)
	at org.lwjgl.system.JNI.invokeV(Native Method)
	at org.lwjgl.glfw.GLFW.glfwPollEvents(GLFW.java:3403)
	at com.mojang.blaze3d.systems.RenderSystem.flipFrame(RenderSystem.java:197)
	at net.minecraft.class_1041.method_15998(class_1041.java:315)
	at net.minecraft.class_310.method_1523(class_310.java:1205)
	at net.minecraft.class_310.method_1514(class_310.java:768)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:527)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
-- Affected level --
Details:
	All players: 1 total; [class_746['Aninuscsalas'/1, l='ClientLevel', x=4.15, y=90.12, z=241.37]]
	Chunk stats: 1024, 255
	Level dimension: minecraft:overworld
	Level spawn location: World: (-80,111,320), Section: (at 0,15,0 in -5,6,20; chunk contains blocks -80,-64,320 to -65,319,335), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Level time: 7048 game time, 7048 day time
	Server brand: quilt
	Server type: Integrated singleplayer server
Stacktrace:
	at net.minecraft.class_638.method_8538(class_638.java:453)
	at net.minecraft.class_310.method_1587(class_310.java:2341)
	at net.minecraft.class_310.method_1514(class_310.java:787)
	at net.minecraft.client.main.Main.method_44604(Main.java:244)
	at net.minecraft.client.main.Main.main(Main.java:51)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.quiltmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:527)
	at org.quiltmc.loader.impl.launch.knot.Knot.launch(Knot.java:82)
	at org.quiltmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:28)
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88)
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126)
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71)
-- Last reload --
Details:
	Reload number: 1
	Reload reason: initial
	Finished: Yes
	Packs: Supplementaries Generated Pack, spoornpacks, Default, Unified Zombies.zip, EBE_v1.1_1.19.x.zip, bedrockwaters/resources, Global Resources, dynamic_asset_generator:assets, purpeille/theinar, Blue Lotus Pack, VanillaTweaks_r733776.zip
-- System Details --
Details:
	Minecraft Version: 1.19.2
	Minecraft Version ID: 1.19.2
	Operating System: Linux (amd64) version 6.3.4-201.fc38.x86_64
	Java Version: 17.0.6, Flathub
	Java VM Version: OpenJDK 64-Bit Server VM (mixed mode, sharing), Flathub
	Memory: 400314808 bytes (381 MiB) / 2097152000 bytes (2000 MiB) up to 2097152000 bytes (2000 MiB)
	CPUs: 4
	Processor Vendor: AuthenticAMD
	Processor Name: AMD Athlon Gold 3150U with Radeon Graphics
	Identifier: AuthenticAMD Family 23 Model 24 Stepping 1
	Microarchitecture: Zen / Zen+
	Frequency (GHz): -0.00
	Number of physical packages: 1
	Number of physical CPUs: 2
	Number of logical CPUs: 4
	Graphics card #0 name: Picasso/Raven 2 [Radeon Vega Series / Radeon Vega Mobile Series]
	Graphics card #0 v
endor: Advanced Micro Devices, Inc. [AMD/ATI] (0x1002)
	Graphics card #0 VRAM (MB): 258.00
	Graphics card #0 deviceId: 0x15d8
	Graphics card #0 versionInfo: unknown
	Virtual memory max (MB): 14818.98
	Virtual memory used (MB): 9875.52
	Swap memory total (MB): 11926.99
	Swap memory used (MB): 4646.17
	JVM Flags: 2 total; -Xms200m -Xmx2000m
	Loaded Shaderpack: (off)
	NEC status: No NEC detected
	Quilted Fabric API: !! WARNING !! This instance is using Fabric API modules re-implemented by QSL. If the issue comes from Quilted Fabric API, DO NOT report to Fabric; report them to Quilt instead!
```

**Comment by aninusmuffin** [comment]
Seems to be a problem on ModernFix side

**Comment by aninusmuffin** [comment]
Fixed in 3.1.0
